demos/ota/ota_demo_core_mqtt/ota_demo_core_mqtt.c

```diff
@@ -0,0 +1,1354 @@
+                                 ( const uint8_t * ) ( CLIENT_IDENTIFIER ),
+                                 otaAppCallback ) ) != OtaErrNone )
+        {
+            LogError( ( "Failed to initialize OTA Agent, exiting = %u.",
+                        otaRet ) );
+
+            returnStatus = EXIT_FAILURE;
+        }
+    }
+
+    /****************************** Create OTA Task. ******************************/
+
+    if( returnStatus == EXIT_SUCCESS )
+    {
+        if( pthread_create( &threadHandle, NULL, otaAgentTask, NULL ) != 0 )
+        {
+            LogError( ( "Failed to create OTA thread: "
+                        ",errno=%s",
+                        strerror( errno ) ) );
+
+            returnStatus = EXIT_FAILURE;
+        }
+    }
+
+    /****************************** OTA Demo loop. ******************************/
+
+    if( returnStatus == EXIT_SUCCESS )
+    {
+        /* Wait till OTA library is stopped, output statistics for currently running
+         * OTA job */
+        while( ( ( state = OTA_GetState() ) != OtaAgentStateStopped ) )
+        {
+            if( mqttSessionEstablished != true )
+            {
+                /* Connect to MQTT broker and create MQTT connection. */
+                returnStatus = establishConnection();
+
+                if( returnStatus == EXIT_SUCCESS )
+                {
+                    mqttSessionEstablished = true;
+
+                    /* Check if OTA process was suspended and resume if required. */
+                    if( state == OtaAgentStateSuspended )
+                    {
+                        /* Resume OTA operations. */
+                        OTA_Resume();
+                    }
+                    else
+                    {
+                        /* Send start event to OTA Agent.*/
+                        eventMsg.eventId = OtaAgentEventStart;
+                        OTA_SignalEvent( &eventMsg );
+                    }
+                }
+            }
+
+            if( mqttSessionEstablished == true )
+            {
+                /* Loop to receive packet from transport interface. */
+                mqttStatus = MQTT_ProcessLoop( &mqttContext, OTA_EXAMPLE_TASK_DELAY_MS );
+
+                /* Get OTA statistics for currently executing job. */
+                OTA_GetStatistics( &otaStatistics );
+
+                if( mqttStatus == MQTTSuccess )
+                {
+                    LogInfo( ( " Received: %u   Queued: %u   Processed: %u   Dropped: %u",
+                               otaStatistics.otaPacketsReceived,
+                               otaStatistics.otaPacketsQueued,
+                               otaStatistics.otaPacketsProcessed,
+                               otaStatistics.otaPacketsDropped ) );
+                }
+                else
+                {
+                    LogError( ( "MQTT_ProcessLoop returned with status = %u.",
+                                mqttStatus ) );
```

Comment:
```suggestion
                    LogError( ( "MQTT_ProcessLoop returned with status = %s.",
                                MQTT_Strerror( mqttStatus ) ) );
```

Reply:
Addressed in https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1418/commits/726a7d066006ec340f26308d37075a414069f56b